bezier-rs/lib/src/lib.rs

```diff
@@ -729,8 +751,57 @@ impl Bezier {
 			return curve.intersections(self, Some(error));
 		}
 
+		// TODO: Considering using the `intersections_between_vectors_of_curves` helper function here
 		// Otherwise, use bounding box to determine intersections
-		self.intersections_between_subcurves([0., 1.], curve, error)
+		self.intersections_between_subcurves([0., 1.], curve, [0., 1.], error).iter().map(|t_values| t_values[0]).collect()
+	}
+
+	/// Helper function to compute intersections between lists of subcurves.
+	/// This function uses the algorithm implemented in `intersections_between_subcurves`.
+	fn intersections_between_vectors_of_curves(subcurves_1: Vec<&(Bezier, [f64; 2])>, subcurves_2: Vec<&(Bezier, [f64; 2])>, error: f64) -> Vec<[f64; 2]> {
+		let segment_pairs = subcurves_1.iter().flat_map(|&(curve_1, curve_1_t_pair)| {
+			subcurves_2.iter().filter_map(move |&(curve_2, curve_2_t_pair)| {
+				if utils::do_rectangles_overlap(curve_1.bounding_box(), curve_2.bounding_box()) {
+					Some((curve_1, curve_1_t_pair, curve_2, curve_2_t_pair))
```

Comment:
I think it would be cleaner to change every `curve_1`/`curve_2` into `curve1` and `curve2` throughout this whole file. Meaning some, like `curve_1_a` and `curve_1_t_pair` would become `curve1_a` and `curve1_t_pair`. That'd make it consistent with other variables around here just called `curve1`.